src/NLog/Config/LoggingConfiguration.cs

```diff
@@ -397,8 +411,10 @@ internal void ValidateConfig()
             // so that whenever the container is initialized its children have already been
             InternalLogger.Info("Found {0} configuration items", this.configItems.Length);
 
-            foreach (object o in this.configItems)
+            var configItemList = this.configItems.ToList();
```

Comment:
`InternalLogger.Info("Found {0} configuration items", this.configItems.Length);`
Might need to move this here. 

Why would you want to convert a array to a list? Since the for can act on a array.